kortex_examples/src/full_arm/example_full_arm_movement.cpp

```diff
@@ -207,30 +199,84 @@ bool example_send_joint_angles(ros::NodeHandle n, const std::string &robot_name,
 {
   last_action_notification_event = 0;
   // Initialize the ServiceClient
-  ros::ServiceClient service_client_play_joint_trajectory = n.serviceClient<kortex_driver::PlayJointTrajectory>("/" + robot_name + "/base/play_joint_trajectory");
-  kortex_driver::PlayJointTrajectory service_play_joint_trajectory;
+  ros::ServiceClient service_client_execute_waypoints_trajectory = n.serviceClient<kortex_driver::ExecuteWaypointTrajectory>("/" + robot_name + "/base/execute_waypoint_trajectory");
+  kortex_driver::ExecuteWaypointTrajectory service_execute_waypoints_trajectory;
+
+  ros::ServiceClient service_client_validate_waypoint_list = n.serviceClient<kortex_driver::ValidateWaypointList>("/" + robot_name + "/base/validate_waypoint_list");
+  kortex_driver::ValidateWaypointList service_validate_waypoint_list;
+
+  kortex_driver::WaypointList trajectory;
+  kortex_driver::Waypoint waypoint;
+  kortex_driver::AngularWaypoint angularWaypoint;
 
-  std::vector<double> angles_to_send;
+  // Angles to send the arm to vertical position (all zeros)
   for (unsigned int i = 0; i < degrees_of_freedom; i++)
   {
-    angles_to_send.push_back(0.0);
+    angularWaypoint.angles.push_back(0.0);
+  }
+
+
+  // The duration parameter for an angularWaypoint must be long enough for the arm to reach its position
+  // so a duration of 0 won't work and ValidateWaypointList will return an error
+  int angular_duration = 0;
+  angularWaypoint.duration = angular_duration;
+
+  // Initialize waypoint
+  waypoint.oneof_type_of_waypoint.angular_waypoint.push_back(angularWaypoint);
+
+  // This duration parameter (for a WaypointList object) can be 0, which means will reach its position in an optimal time
+  // However, this parameter is overriden by the AngularWaypoint duration
+  trajectory.duration = 0;
+  trajectory.use_optimal_blending = 0;
+  trajectory.waypoints.push_back(waypoint);
+
+  // ValidateWaypointList will return an error if the robot does not have time to 
+  // reach its position so we can use it to find an almost optimal duration by 
+  // incrementing duration until we do not get any error
```

Comment:
There is something like 4 comment blocks to explain some things related to `duration` parameter, can we make it one and make it very clear?
e.g. `For Angular Waypoints, you have to put a duration for each wp and the global duration is disregarded. For Cartesian Waypoints, the duration for 1 wp is ignored. If you put 0 global duration, the trajectory will be optimal and if you put somehting too small, the trajectory will be rejected.`

Reply:
done